pkg/server/plugin/noderesolver/azure/client.go

```diff
@@ -58,6 +58,13 @@ func (c *azureClient) GetVirtualMachineResourceID(ctx context.Context, principal
 	}
 
 	values := result.Values()
+	for len(values) == 0 {
```

Comment:
This usage of the result doesn't look quite right. I would not expect the first page of values to be empty if there were multiple pages of results.

This also obscures the error case when no values are returned (handled in the next `if` block) by returning a more generic error from the Azure SDK from `result.NextWithContext()`.

Reply:
Azure SDK sometimes gives you an empty response with page id if the num of the VMs is more than 1000.
so to handle this you have to traverse to different pages as well.

Currently, this is not taken care of by the Azure SDK.

Reply:
So if you are getting the next page id in the response, it means that there is some data on the next page.

Reply:
@mayur040993 do you have a link to the relevant documentation from the Azure APIs that describe this behavior?